packages/react/ds/src/text-input/text-input.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+// Extend `React.InputHTMLAttributes<HTMLInputElement>` so that
+// the component can accept all the standard attributes and events that an `<input>` element can handle.
+export type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
+  ref?: React.Ref<HTMLInputElement>;
+  hasError?: boolean;
+  prefix?: React.ReactNode;
+  suffix?: React.ReactNode;
+  halfFluid?: boolean;
+  fullFluid?: boolean;
+  characterWidth?: number;
+};
+
+// Use React.forwardRef to support refs properly
+export const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
+  (
+    {
+      hasError = false,
+      prefix,
+      suffix,
+      halfFluid,
+      fullFluid = true,
+      characterWidth,
+      ...props
+    },
+    ref,
+  ) => {
+    // Determine dynamic width class
+    const widthClass =
+      fullFluid && !halfFluid && !characterWidth
+        ? 'gi-w-full'
+        : halfFluid
+          ? 'gi-w-1/2'
+          : characterWidth
+            ? // TODO Not working. Review it with the team.
+              `gi-w-[${characterWidth}em]`
```

Comment:
@pregno @BogdanDanilescu 

Can you please help me with this? E.g. if `characterWidth = 4` it will result in `gi-w-[4em]` to the `div` but somehow this not reflecting.

Reply:
That's most probably because the classes are not generated at build time see here https://tailwindcss.com/docs/content-configuration#classes-arent-generated

Anyway, why are you using `[4em]` and not `w-16`

Reply:
I think it's because the class is not being generated since you are using `gi-w-[${characterWidth}em]` and it doesn't know which class to generate. One way to get around this is to safelist the classes. 

https://tailwindcss.com/docs/content-configuration#safelisting-classes

This ensures that whatever classes we add to the safelist object, it will be available in the application at runtime